src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -864,24 +868,44 @@ def apply_lifestyle_ramp(self, pop_visitors: pd.DataFrame) -> pd.DataFrame:
             & (fpg_not_tested_recently)
         )
 
-        # Determine which eligible simulants get assigned a test date
+        # Determine which simulants eligible for FPG testing actually get tested
         tested_simulants = self.randomness.filter_for_probability(
             pop_visitors[is_eligible_for_testing],
             [data_values.FPG_TESTING.PROBABILITY_OF_TESTING_GIVEN_ELIGIBLE]
             * sum(is_eligible_for_testing),
         )
 
+        # Test simulants for FPG
         pop_visitors.loc[
             tested_simulants.index, data_values.COLUMNS.LAST_FPG_TEST_DATE
         ] = self.clock()
 
+        fpg = self.fpg(tested_simulants.index)
+
+        # Enroll in lifestyle by updating lifestyle column with date of enrollment
+        fpg_within_bounds = (fpg >= data_values.FPG_TESTING.LOWER_ENROLLMENT_BOUND) & (
+            fpg <= data_values.FPG_TESTING.UPPER_ENROLLMENT_BOUND
+        )
+        enroll_if_fpg_within_bounds = (
+            self.lifestyle(tested_simulants.index) == data_values.LIFESTYLE_EXPOSURE.EXPOSED
+        )
+        newly_enrolled = fpg_within_bounds & enroll_if_fpg_within_bounds
+        tested_simulants.loc[newly_enrolled, data_values.COLUMNS.LIFESTYLE] = self.clock()
+
         self.population_view.update(
             pop_visitors[
                 [
                     data_values.COLUMNS.LAST_FPG_TEST_DATE,
                 ]
             ]
         )
+        self.population_view.update(
```

Comment:
Nit: make the changes to the Lifestyle enrollment column directly on `pop_visitors` so you only need to make one `update` call.

Reply:
That was the cleaner approach I mentioned I was going to spend 15 minutes on this morning, but some of the booleans would only get defined properly using the tested simulants subset so I just went with this since it's working.